packages/xy-chart/src/selection/Brush.jsx

```diff
@@ -95,24 +116,64 @@ class Brush extends React.Component {
   convertRangeToDomain(brush) {
     const { xScale, yScale } = this.props;
     const { x0, x1, y0, y1 } = brush.extent;
-    const invertedX0 = xScale.invert(x0 + (x0 < x1 ? -SAFE_PIXEL : SAFE_PIXEL));
-    const invertedX1 = xScale.invert(x1 + (x1 < x0 ? -SAFE_PIXEL : SAFE_PIXEL));
-    const invertedY0 = yScale.invert(y0 + (y0 < y1 ? -SAFE_PIXEL : SAFE_PIXEL));
-    const invertedY1 = yScale.invert(y1 + (y1 < y0 ? -SAFE_PIXEL : SAFE_PIXEL));
+
+    let xDomain;
```

Comment:
I see what you mean about duplicate code, could this be pulled out into a helper function that does the logic that's duplicated for x + y now?

I think you're right that this will have to return an array of values for a band scale.